federation/executor.go

```diff
@@ -159,10 +274,10 @@ func (e *Executor) execute(ctx context.Context, p *Plan, keys []interface{}) (in
 			resMu.Lock()
 			defer resMu.Unlock()
 			for k, v := range result {
-				if _, ok := subPlanMetaData.results[k]; ok {
-					return oops.Errorf("key already exists in results: %v", k)
+				if _, ok := subPlanMetaData.results[k]; !ok {
+					subPlanMetaData.results[k] = v
 				}
```

Comment:
Sometimes multiple responses will have the result for the field "__key" which leads to conflicts. I changed it so if it sees a key conflict for a field that is not "__key" it will show an error.

Reply:
should the response only contains one `__key` field?

Reply:
We talked about thsi in person, but to document the example that illustrates the `__key` conflict.If we have the query 
```
{
    users {
        id
        email
     }
}
```

We split it into two queries 
```
{
  __federation {
    users(keys) {
       id
    }
}

{
  __federation {
    users(keys) {
       email
    }
}
```

which would return two results 
```
{
  user {
      __key
      email
  }
}
```
and 
```
{
  user {
      __key
      id
  }
}
```

leading to a conflict for key.